osconfig_tests/compute/instance.go

```diff
@@ -38,16 +39,17 @@ func (i *Instance) Cleanup() {
 	}
 }
 
-// WaitForSerialOutput waits to a string match on a serial port.
-func (i *Instance) WaitForSerialOutput(match string, port int64, interval, timeout time.Duration) error {
+// WaitForSerialOutput waits for a match with regular expression regex on a serial port.
+func (i *Instance) WaitForSerialOutput(regex string, port int64, interval, timeout time.Duration) error {
 	var start int64
 	var errs int
+	var validMatch = regexp.MustCompile(regex)
```

Comment:
Use compile and then check for and return the error.

Reply:
changed it